src/main/java/net/mcreator/element/parts/gui/Button.java

```diff
@@ -50,12 +52,13 @@ public Button(String name, int x, int y, String text, int width, int height, Pro
 	}
 
 	@Override public void paintComponent(int cx, int cy, WYSIWYGEditor wysiwygEditor, Graphics2D g) {
-		g.drawImage(MinecraftImageGenerator.generateButton(this.width, this.height), cx, cy, this.width, this.height,
-				wysiwygEditor);
+		if (!isUndecorated)
+			g.drawImage(MinecraftImageGenerator.generateButton(this.width, this.height), cx, cy, this.width,
+					this.height, wysiwygEditor);
 		int textwidth = (int) (WYSIWYG.fontMC.getStringBounds(this.text, WYSIWYG.frc).getWidth());
 		int textheight = (int) (WYSIWYG.fontMC.getStringBounds(this.text, WYSIWYG.frc).getHeight()) - 4;
-		g.drawString(this.text, cx + (this.width / 2) - (textwidth / 2),
-				cy + textheight + (this.height / 2) - (textheight / 2));
+		g.drawString(this.text, isUndecorated ? cx : cx + (this.width / 2) - (textwidth / 2),

```

Comment:
The code would be easier to read if you split this into two different drawString methods and split it with if statement for the two cases possible